src/shared/Microsoft.Git.CredentialManager/HttpClientFactory.cs

```diff
@@ -85,6 +87,35 @@ public HttpClient CreateClient()
 #endif
             }
 
+            string sslCaInfo = _settings.SslCaInfo;
+            if (!string.IsNullOrWhiteSpace(sslCaInfo))
+            {
+                Func<X509Chain, SslPolicyErrors, bool> validationCallback = (chain, errors) =>
+                {
+                    if ((errors & ~SslPolicyErrors.RemoteCertificateChainErrors) != 0 ||
+                        chain.ChainStatus.Any(status => status.Status != X509ChainStatusFlags.UntrustedRoot))
+                    {
+                        // Fail validation if remote cert has any errors other than untrusted root
+                        return false;
+                    }
+
+                    // Get the custom CA root from http.sslCAInfo
+                    X509Certificate2 caBundle = new X509Certificate2(sslCaInfo);
```

Comment:
`X509Certificate2` [implements `IDisposable` starting from .NET Framework 4.6](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509certificate2?view=netframework-4.7.2#remarks), so we should wrap this in a `using` block (or call `Dispose` in the `finally` block of a `try-finally`).

It might also be good measure to test what the happens if an exception is thrown in this validation callback; does it go unhandled?

Reply:
> X509Certificate2 implements IDisposable starting from .NET Framework 4.6, so we should wrap this in a using block (or call Dispose in the finally block of a try-finally).

Will do!

> It might also be good measure to test what the happens if an exception is thrown in this validation callback; does it go unhandled?

The "non-existent file" test I noted actually leads to an uncaught exception (the `X509Certificate2` constructor reads the file from disk - if it's not found, an exception is thrown. The result is this output:

```
fatal: The SSL connection could not be established, see inner exception.
fatal: Could not find file '<path to the fake cert>'.
```

Reply:
Update on this - because the file is now imported as an `X509Certificate2Collection` (since it can contain multiple certs), it's no longer an `IDisposable` type. Should the collection elements be manually disposed (e.g., in a loop), or left without manual disposal?